design/ceph/object/ceph-rgw-k8s-sa-authentication.md

```diff
@@ -0,0 +1,66 @@
+---
+Service authentication with vault for RGW
+target-version: release-1.10
+---
+
+# Feature Name
+Service account authentication with vault for RGW 
+
+## Summary
+When `Vault` is set up in Kubernetes environment, it can be authenticated using service account token. The OSD encryption already supports this feature and adding same for RGW. It uses sameset of  config option in `ConnectionDetails` for `KeyManagementServiceSpec`.
+
+### Goals
+Configure RGW with service account authentication with Vault with help of vault agent.
+
+### Non-Goals
+There are different ways RGW can authenticate with help of [vault agent](https://docs.ceph.com/en/latest/radosgw/vault/#vault-agent), but only service account authentication is supported here.
+
+## Proposal details
+The service account details will specified in `ConnectionDetails` of `KeyManagementServiceSpec`, using that info plus the other details a config map with <ceph-object-store>-vault-agent-cm populated for the vault agent sidecar container. The side container details will be added RGW pod spec and start sidecar container with RGW pod. The RGW will be configured with vault agent specific options. For the following details in `ConnectionDetails`:
```

Comment:
Is this sidecar required? It is non-ideal for Rook to hard-code support for Vault, especially in today's world where the vault injector is available. Why don't we instruct users to use the injector instead?

Reply:
> Is this sidecar required? It is non-ideal for Rook to hard-code support for Vault, especially in today's world where the vault injector is available. Why don't we instruct users to use the injector instead?

Vault injector Job is different, it does not authenticate with applications. It just inject  vault secret directly to application pod as file kinda of webhook